zap/src/output/luau/client.rs

```diff
@@ -366,27 +435,39 @@ impl<'src> ClientOutput<'src> {
 				self.push_line(&format!("events[{id}] = {{}}"));
 			}
 
+			if ev_decl.data.is_some() {
+				self.push_line(&format!("event_queue[{id}] = {{}}"));
+			} else {
+				self.push_line(&format!("event_queue[{id}] = 0"));
+			}
+		}
+
+		for (i, ..) in self.config.fndecls.iter().enumerate() {
+			let id = i + 1;
```

Comment:
`i` cannot be used as a means of getting event id because it will intersect with evdecls

Reply:
Should we assign in ID in `convert.rs` instead?

Reply:
sgtm